Makefile

```diff
@@ -12,41 +12,80 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-include include.mk
-
-SUBDIRS := expandybird/. resourcifier/. manager/.
-TARGETS := all build test push container clean
-
-SUBDIRS_TARGETS := \
-	$(foreach t,$(TARGETS),$(addsuffix $t,$(SUBDIRS)))
+ifndef GOPATH
+$(error No GOPATH set)
+endif
 
-GO_DEPS := github.com/kubernetes/deployment-manager/util/... github.com/kubernetes/deployment-manager/version/... github.com/kubernetes/deployment-manager/expandybird/... github.com/kubernetes/deployment-manager/resourcifier/... github.com/kubernetes/deployment-manager/manager/... github.com/kubernetes/deployment-manager/dm/...
+include include.mk
 
-.PHONY : all build test clean $(TARGETS) $(SUBDIRS_TARGETS) .project .docker
+GO_DIRS ?= $(shell glide nv -x )
+GO_PKGS ?= $(shell glide nv)
 
+.PHONY: build
 build:
-	go get -v $(GO_DEPS)
-	go install -v $(GO_DEPS)
+	@scripts/build-go.sh
```

Comment:
This is a substantial change to the build process. Would appreciate a walk through.